README.md

```diff
@@ -1 +1,123 @@
+  - downloads one of the associated archives (and perhaps extracts its `PackageInfo.g`, and verifies it has a matchin checksum?)
+    - if possible we want .tar.gz; in case any package doesn't have one, we may need a script to generate one (e.g. tar.bz2 is trivial, not sure if there is a package that only offers .zip? perhaps we should *require* a `.tar.gz`???)
+  - computes checksums of PI.g and the archive, adds them to the metadata
+  - writes out the metadata as JSON
+- alternate usecase: take the name of a .tar.gz as argument;
+  extract its `PackageInfo.g`, then proceed similar to before?!?
+
+
+### `upload-package.py`
+- to upload such a package tarball to GitHub release (could use [GitHub's `gh` tool](http://cli.github.com) or adapt the code)
+- also would be nice to be able to upload to our own server, but that'll probably amount to a `scp FILE.tar.gz ourserver:/some/path` done in a GH action, which gets a suitable SSH prviate key (granting it write access) via a repository secreet
+
+
+### `assemble-distro.py`
+- scans over all metadata, collects the appropriate tarballs, then "merges" them into one big tarball
+- the assembled distro archive also needs to be uploaded, but ideally we can just repurpose `upload-package.py`
+- should do caching: download tarballs into a specific fixed location; if a file with the right name already exists and has the right checksum, just use it
+
+
+## Automation using GitHub Actions
+
+The scripts from the previous section are called from GitHub Actions
+to automate most of the maintenance on the GAP package distribution.
+This works as follows:
+
+### Cron job on `PackageDistro`
+
+A GH action that runs regulalry and updates PRs for package updates,
+one per package (if someone needs to update multiple packages in one
+PR, they have to submit a PR manually)
+
+- This CI job is called regularly, e.g. once per hour or day or so
+- It invokes the `scan-for-updates.py` script
+- it then loops over the reported `PackageInfo.g` files, and for each...
+  - check if an update branch already exists, say `update/pkgname`
+     - if it does, check it out
+        - if the metadata differs, then update the branch and force push it
+        - check if a PR is still open for that branch; if not, open one        
+          (this deals with the case when a PR is accidentally removed/close/not created
+           due to a hiccup before. Note that this means: if maintainers don't want a
+           certain package release for whatever reason: don't close that PR, leave
+           it open (possibly marked with a label)
+     - if no such branch exists
+        - create a new branch
+        - update it with the new metadata
+        - push it
+        - open a new PR
+
+TODO TODO TODO: actually there is a mismatch here between what this CI job is
+supposed to do and what the `import-package.py` script does. Seems we need to
+parse the `PackageInfo.g` files in here as well, to "update the JSON"
```

Comment:
@james-d-mitchell FYI. I've just discussed this with @ThomasBreuer . My plan would be this: I do need to parse the PackageInfo.g here, but I'll not do anything else, other than printing it into the appropriate `meta.json` file, together with the SHA256... (errr .... and of course I need the SHA256 of the archive as well... so I guess I kinda have to download it here... so we'll download it at least twice overall... not the most elegant, but I can't think of a better way?)

Anyway, then (as described above) create a branch+commit+PR with that new meta.json. The CI for that PR (discussed below) will then detect which meta.json file(s) were modified, and invoke `import-package.py` suitably. But that means `import-package.py` will actually "see" a meta.json that's already for the new version... That's different from what we discussed so far... But shouldn't be too hard to deal with, right?